app/controllers/payment-links/metadata/metadata-form.js

```diff
@@ -80,4 +99,17 @@ function isNotEmpty (value) {
   return value && value.length !== 0
 }
 
+function isValidLength (value) {
+  return value && value.length !== 0 && value.length <= META_DATA_MAX_LENGTH
+}
+
+function isNotDuplicate (value, existingMetadata) {
+  if (existingMetadata && existingMetadata.length > 0) {
```

Comment:
is `existingMetadata` expecting an array with a length? I think the format of metadata in general is an object of key value pairs, like
```js
{ key: value, anotherKey: value }
```

Is it possible to update this method to work with that? I don't know of anywhere where we have an array of objects with `key`, `value` params, potentially this could use something like
```js
const found = Object.keys(existingMetadata).find(metadataKey => value === metadataKey)
```

Reply:
Done